src/linear_algebra/free_module/strong_rank_condition.lean

```diff
@@ -0,0 +1,53 @@
+/-
+Copyright (c) 2021 Riccardo Brasca. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Riccardo Brasca
+-/
+
+import linear_algebra.charpoly.basic
+import linear_algebra.invariant_basis_number
+
+/-!
+
+# Strong rank condition for commutative rings
+
+We prove that any nontrivial commutative ring satisfies `strong_rank_condition`.
+
+## Main result
+
+* `comm_ring_strong_rank_condition R` : `R` has the `strong_rank_condition`.
+
+## References
+
+We follow the proof given in https://mathoverflow.net/a/47846/7845.
+
+-/
+
+variables (R : Type*) [comm_ring R] [nontrivial R]
+
+open polynomial function fin nat linear_map
+
+instance comm_ring_strong_rank_condition : strong_rank_condition R :=
+begin
+-- It is enough to prove that for all `n`, any `f: (fin (n + 1) → R) →ₗ[R] fin n → R`
+-- is not injective. By contradiction, we assume that `f` is injective.
+  apply (strong_rank_condition_iff_succ R).2 (λ n f hf, _),
```

Comment:
If you want to explain the proof, why not use verbose tactics rather than comments? For instance, this could become:
```suggestion
  suffices : ∀ n, ∀ f : (fin (n + 1) → R) →ₗ[R] fin n → R, ¬injective f,
  { rwa strong_rank_condition_iff_succ R },
  intros n f, by_contradiction hf,
```